### PR TITLE
[NTRN-163] add dex stargate tests

### PR DESCRIPTION
**Pull request by sotnikov-s**
**task:** https://hadronlabs.atlassian.net/browse/NTRN-163

the code is a week testing of stargate queries and messages to the dex module made mostly to make sure all the messages and queries are accepted and processed without thorough check of the result values. the code is mostly a copy+paste from here: https://github.com/neutron-org/neutron-integration-tests/pull/240 with a few changes/adaptations

**related PRs:**
- https://github.com/neutron-org/neutron-sdk/pull/121
- https://github.com/neutron-org/neutron-dev-contracts/pull/37
- https://github.com/neutron-org/neutronjsplus/pull/12

### CODE REVIEW

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7111070219

File: src/testcases/run_in_band/dex_stargate.test.ts [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+  LimitOrderTrancheResponse,
+  LimitOrderTrancheUserResponse,
+  LimitOrderType,
+  ParamsResponse,
+  PoolMetadataResponse,
+  PoolReservesResponse,
+  PoolResponse,
+} from '../../helpers/dex';
+
+describe('Neutron / IBC hooks', () => {
```
Comment (swelf19) [inline src/testcases/run_in_band/dex_stargate.test.ts:30]:
```suggestion
describe('Neutron / dex module (stargate contract)', () => {
```

File: src/testcases/run_in_band/dex_stargate.test.ts [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+      testState.blockWaiter1,
+      NEUTRON_DENOM,
+    );
+    neutronAccount = new WalletWrapper(
+      neutronChain,
+      testState.wallets.neutron.demo1,
+    );
+  });
+
+  describe('Instantiate hooks ibc transfer contract', () => {
```
Comment (swelf19) [inline src/testcases/run_in_band/dex_stargate.test.ts:52]:
```suggestion
  describe('Instantiate dex stargate contract', () => {
```

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7111070219

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7206923966

**Review by pr0n00gler — DISMISSED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Review by swelf19 — APPROVED** [review]

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/7206923966

**Pull request merged**